src/env.cc

```diff
@@ -671,6 +671,36 @@ void Environment::PrintSyncTrace() const {
                       isolate(), stack_trace_limit(), StackTrace::kDetailed));
 }
 
+MaybeLocal<Value> Environment::RunSnapshotSerializeCallback() const {
+  if (!snapshot_serialize_callback().IsEmpty()) {
```

Comment:
The persistent handle `snapshot_serialize_callback` is being snapshotted too so the callback can be called even when it is not building snapshots. Though the run may have no effects since the callbacks are cleared out after being invoked. Can we have a short path to prevent unnecessary calling into JS after deserialization?

Reply:
Done